tools/detekt/src/test/kotlin/com/datadog/tools/detekt/rules/InvalidPerfMeasureBlockTest.kt

```diff
@@ -0,0 +1,339 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.tools.detekt.rules
+
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import io.github.detekt.test.utils.KotlinCoreEnvironmentWrapper
+import io.github.detekt.test.utils.createEnvironment
+import io.gitlab.arturbosch.detekt.test.assertThat
+import io.gitlab.arturbosch.detekt.test.lintWithContext
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(ForgeExtension::class)
+internal class InvalidPerfMeasureBlockTest {
+
+    lateinit var kotlinEnv: KotlinCoreEnvironmentWrapper
+
+    @BeforeEach
+    fun setup() {
+        kotlinEnv = createEnvironment()
+    }
+
+    @AfterEach
+    fun tearDown() {
+        kotlinEnv.dispose()
+    }
+
+    // region Tests
+
+    @Test
+    fun `M detekt invalid measure block { forged parameters }`() {
+        // Given
+        val code =
+            """ 
+                package com.datadog.android.nightly.rum
+
+                import com.datadog.android.nightly.utils.measure
+                import com.datadog.android.rum.GlobalRum
+                import com.datadog.android.nightly.utils.Forge
```

Comment:
I am not asserting the `Forge` package in the rule, the only reason why I am using it here is to test some real scenarios in our code. If you are asking for the case:
```
measure {
    forge.aBool()
}
```
you will be able to actually measure that because the only thing I am checking is the way the parameters of the function are declared (resolved)

Reply:
Wait if I understand correctly, your rule will raise an issue for : 
```
                    measure { 
                       GlobalRum.get().startView(forge.aKey(), forge.aName())
                    }
```

but it won't raise any warning for 

```
                    measure { 
                       val key = forge.aKey()
                       val name = forge.aName()
                       GlobalRum.get().startView(key, name)
                    }
```

IMHO, then this rule doesn't help much, since you can have a lot of false negatives, wdyt?

Reply:
Damn it, you are right that's a case I did not have in mind as for now we are not using this pattern in our E2E tests. I will try to adapt the rule for this case also, it will be tricky though as we do not know which method is for testing and which one for generating the arguments :(